CMakeLists.txt

```diff
@@ -0,0 +1,214 @@
+
+cmake_minimum_required(VERSION 3.4.3)
+
+list(APPEND CMAKE_MODULE_PATH
+     "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
+
+project(Foundation
+        LANGUAGES
+          C)
+
+option(FOUNDATION_PATH_TO_LIBDISPATCH_SOURCE "Path to libdispatch source" "")
+option(FOUNDATION_PATH_TO_LIBDISPATCH_BUILD "Path to libdispatch build" "")
+
+include(SwiftSupport)
+include(GNUInstallDirs)
+include(ExternalProject)
+
+ExternalProject_Add(CoreFoundation
```

Comment:
The libCoreFoundation.a needs to be a whole archive linkage iirc

Reply:
Interesting, I didnt see that flag used.  But, that should be possible to do.  We explicitly build it static here to allow us to incorporate it completely.